ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -710,7 +715,14 @@ private boolean verifyJoin(
             ));
           }
 
-          return true;
+          if (!(leftExpression instanceof ColumnReferenceExp)) {
+            throw new KsqlException(String.format(
+                "Invalid join condition: foreign-key table-table joins with expressions "
+                    + "are not supported. Got %s = %s."
```

Comment:
You need to pass in `leftExpression` and `rightExpression` into `format()` method